core/src/main/java/org/elasticsearch/index/query/GeoShapeQueryParser.java

```diff
@@ -104,6 +106,8 @@ public GeoShapeQueryBuilder fromXContent(QueryParseContext parseContext) throws
                                     } else if (parseContext.parseFieldMatcher().match(currentFieldName, SHAPE_PATH_FIELD)) {
                                         shapePath = parser.text();
                                     }
+                                } else {
+                                    throw new ParsingException(parser.getTokenLocation(), "[geo_shape] unknown token [" + token + "] after [" + currentFieldName + "]");
```

Comment:
You switched to using the query builder NAME in other places - why not here as well?